src/control_overlay/app.rs

```diff
@@ -1,57 +1,74 @@
 use std::sync::mpsc::Sender;
 
-use leptos::{
-    component, create_signal, event_target_value, IntoView, ReadSignal, SignalGet,
-    SignalSet, view,
-};
+use leptos::{CollectView, component, create_effect, create_signal, IntoView, Memo, ReadSignal, SignalGet, SignalGetUntracked, SignalSet, SignalUpdate, view, WriteSignal};
 
 use crate::control_overlay::action::Action;
-use crate::fabric::interval::Interval;
+use crate::control_state::ControlState;
+use crate::fabric::interval::{Interval, Material, Role, Span};
 
 #[component]
 pub fn ControlOverlayApp(
+    fabric_list: Memo<Vec<String>>,
+    materials: Memo<[Material; 5]>,
     control_state: ReadSignal<ControlState>,
+    set_control_state: WriteSignal<ControlState>,
     actions_tx: Sender<Action>,
 ) -> impl IntoView {
-    let pre_text = move || format!("{:#?}", control_state.get());
-    let (category, set_category) = create_signal("Art".to_string());
-    let (subname, set_subname) = create_signal("Halo by Crane".to_string());
-    let fabric_name = move || vec![category.get(), subname.get()];
-    let on_run_fabric_click = move |_ev| {
-        actions_tx
-            .send(Action::LoadFabric(fabric_name()))
-            .expect("failed to send action");
+    let (name, set_name) = create_signal("".to_string());
+    create_effect(move |_| {
+        if !name.get().is_empty() {
+            set_control_state.update(|state| *state = ControlState::Viewing);
+            actions_tx
+                .send(Action::LoadFabric(name.get()))
```

Comment:
could extract name. i like doing sth like this 

```
let name = name.get();
```

because you can shadow in rust